tests/unit/s2n_mem_usage_test.c

```diff
@@ -13,6 +13,15 @@
  * permissions and limitations under the License.
  */
 
+
+#ifdef __FreeBSD__
+/* FreeBSD requires POSIX compatibility off for its syscalls (enables __BSD_VISIBLE)*/
+#undef _POSIX_C_SOURCE
```

Comment:
Can you expand on this a little more? What's the problem?

Reply:
The long story short, if you include <sys/types.h> with _POSIX_C_SOURCE defined, it sets __BSD_VISIBLE to 0. Which in turn prevents the BSD types from being defined that are needed to make FreeBSD sysctl calls. The alternative is that I could set up linprocfs in our CI/CD pipelines, but this isn't common and could result in confusion between running the tests locally vs running them in the pipelines. I'll add a little more context to the comment here.